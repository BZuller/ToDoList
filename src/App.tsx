import { useEffect, useState } from "react";
import TodoTask from "./components/TodoTask/TodoTask";
import { ITask } from "./Interfaces";

import './styles/styles.css'
import axios from "axios";

function App() {


	const [task, setTask] = useState<string>("")

	const [todoList, setTodoList] = useState<ITask[]>([])

	useEffect(() => {
		axios.get(`http://localhost:3333/tasks`)
		.then(resposta=> setTodoList(resposta.data))
		.catch(()=> console.log('Nao gerou a array'))
	}, [todoList]
	)

	const [filter, setFilter] = useState<ITask[]>(todoList)

    useEffect(() => setFilter(todoList), [todoList])

	function addTask(name: String, description: String) {
		axios.post(`http://localhost:3333/tasks`, {
			name, description
		})
		.then((response)=> console.log(response))
		.catch(()=> console.log('Erro ao adicionar a task!'))
	}


	function verPendentes () {
					return setFilter(todoList.filter((tarefa: ITask) => tarefa.complete === false))
				}

	function verFinalizadas () {
					return setFilter(todoList.filter((tarefa: ITask) => tarefa.complete === true))
				}
	function verTodas(){
	 return setFilter(todoList)
	}

	function DeleteTask(DeleteTaskById: string) {
		axios.delete(`http://localhost:3333/tasks/${DeleteTaskById}`)
		.then(() => {console.log('Deletado')})
		.catch(() => {console.log('Erro inesperado!')})
	}

	return (
		<div className="App">

			<header>

				<h2>Lista de tarefas</h2>

				<input
					type="text" autoComplete="off"
					placeholder="Digite sua tarefa..."
					name="task"
					className="input"
					value={task}
					onChange={(event) => setTask(event.target.value)}
					required
				/>
				<div className = "div-filters">
				<button type="submit" className="btn-header" onClick={()=>addTask(task, task)}>Adicionar Tarefa</button>

				<button type="submit" className="filters" onClick={() => verFinalizadas()}>Apenas concluidas</button>

				<button type="submit" className="filters" onClick={() => verPendentes()}>Apenas pendentes</button>

				<button type="submit" className="filters" onClick={() => verTodas()}>Mostrar todas</button>
				</div>
			</header>
			<div className="line"></div>

			{filter.map((task, key) => (
			<TodoTask todoList={todoList} key={key} task = {task} deleteTask={DeleteTask} setTodoList={setTodoList} filter={filter}/>
			))}
		</div>
	);
}

export default App;
