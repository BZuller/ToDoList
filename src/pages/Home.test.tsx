import React from 'react';
import renderer from 'react-test-renderer';

import App from './Home';

it('renders correctly when there are no tasks', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});