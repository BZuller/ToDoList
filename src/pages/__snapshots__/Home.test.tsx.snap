// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no tasks 1`] = `
<div
  className="App"
>
  <header>
    <h2>
      Lista de tarefas
    </h2>
    <input
      autoComplete="off"
      className="input"
      name="task"
      onChange={[Function]}
      placeholder="Digite sua tarefa..."
      required={true}
      type="text"
      value=""
    />
    <div
      className="div-filters"
    >
      <button
        className="btn-header"
        onClick={[Function]}
        type="submit"
      >
        Adicionar Tarefa
      </button>
      <button
        className="filters"
        onClick={[Function]}
        type="submit"
      >
        Apenas concluidas
      </button>
      <button
        className="filters"
        onClick={[Function]}
        type="submit"
      >
        Apenas pendentes
      </button>
      <button
        className="filters"
        onClick={[Function]}
        type="submit"
      >
        Mostrar todas
      </button>
    </div>
  </header>
  n
  <div
    className="line"
  />
</div>
`;
